junifer/data/template_spaces.py

```diff
@@ -89,3 +92,63 @@ def get_xfm(
                     f.write(chunk)
 
     return xfm_file_path
+
+
+def get_template(
+    space: str,
+    target_data: Dict[str, Any],
+    extra_input: Optional[Dict[str, Any]] = None,
+) -> nib.Nifti1Image:
+    """Get template for the space, tailored for the target image.
+
+    Parameters
+    ----------
+    space : str
+        The name of the template space.
+    target_data : dict
+        The corresponding item of the data object for which the template space
+        will be loaded.
+    extra_input : dict, optional
+        The other fields in the data object. Useful for accessing other data
+        types (default None).
+
+    Returns
+    -------
+    Nifti1Image
+        The template image.
+
+    Raises
+    ------
+    ValueError
+        If ``space`` is invalid.
+    RuntimeError
+        If template in the required resolution is not found.
+
+    """
+    # Check for invalid space; early check to raise proper error
+    if space not in tflow.templates():
+        raise_error(f"Unknown template space: {space}")
+
+    # Get the min of the voxels sizes and use it as the resolution
+    target_img = target_data["data"]
+    resolution = np.min(target_img.header.get_zooms()[:3]).astype(int)
+
+    logger.info(f"Downloading template {space} in resolution {resolution}")
+    # Retrieve template
+    try:
+        template_path = tflow.get(
```

Comment:
We need to get the "closest resolution" here

Reply:
The `closest_resolution` function takes a list of valid resolutions, but we don't have that info right away. For that we need an extra scraping of the templateflow metadata which adds a bit of overhead. I don't get the complete rationale, maybe I can implement it in a different way if I understand correctly.

Reply:
The idea of that function is to tell you which is the resolution of the parcellation/mask/etc that you need to use for the image you have.

Reply:
We are already getting the resolution for the template required from the target data and it needs to be int for templateflow. If templateflow doesn't have that, it throws an error. So in that case, we need to resample.

Reply:
But in our case, we have it and we fetch accordingly.

Reply:
no, we don't have "any" resolution. We have some resolutions. Templateflow is the same. We have A, B, C and D as resolutions. Let's say our image is in resolution E. We need to get the "closest" to E following the criteria defined in the "closest resolution" function.

Reply:
So there are two options:

- scrape templateflow metadata to get updated info but has overhead
- define the possible resolutions for all templateflow-provided templates

Reply:
Just to put it here, `closest_resolution` is used in functions like `load_parcellation` and `load_mask` where one can choose the resolution, but in functions like `get_parcellation` and `get_mask`, we use the target's resolution to resample the data obtained via `load_*` functions.

Reply:
Junifer needs to able to run without network connections. My take is to scrape on the first time and save in the `junifer_data` directory (if we follow the scrape option).

My issue with hard-coding the resolutions is what might happen if templateflow adds a new resolution. We might need to update the code.

Reply:
Templateflow does the caching so should be able to work without network connection. To make this clean and work in a global way, it's better to have something like a `junifer precache` command which downloads required data for datagrabber, parcellations, masks and template spaces.

Reply:
100% agree with the `precache`. I though of calling it `prefetch`. And that's why everything that is donwloaded (like parcellations) are stored in the `junifer_data` directory, to make it explicit and permanent across nodes/sessions/users.

In this case, check which of the above solutions works best in your opinion.

Reply:
I would want to make a separate PR tackling this specific feature and come back to `get_template` then. What do you think?

Reply:
Do you mean `precache`? Yes, that's a totally separate PR.

Reply:
Yes.

For this PR, the scraping won't be optimal until we have precache.

Reply:
First we make it work, then we optimise.

Reply:
The latest commits should address your concern.